src/helpers.c

```diff
@@ -1180,3 +1180,33 @@ bool check_mix_exists(void)
 	free_string(&fullpath);
 	return ret;
 }
+
+/* Iterate the file list and remove from the file system each file/directory */
+void remove_files_from_fs(struct list *files)
+{
+	struct list *iter = NULL;
+	struct file *file = NULL;
+	char *fullfile = NULL;
+	int total = list_len(files);
+	int deleted = total;
+	int count = 0;
+
+	iter = list_head(files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+		string_or_die(&fullfile, "%s/%s", globals.path_prefix, file->filename);
+		if (swupd_rm(fullfile) == -1) {
+			/* if a -1 is returned it means there was an issue deleting the
+			 * file or directory, in that case decrease the counter of deleted
+			 * files.
+			 * Note: If a file didn't exist it will still be counted as deleted,
+			 * this is a limitation */
+			deleted--;
+		}
+		free_string(&fullfile);
+		count++;
+		progress_report(count, total);
+	}
+	info("Total deleted files: %i\n", deleted);
```

Comment:
Don't print this here. Return an int in this function with the total of deleted files and print where appropriate.

Reply:
yeah, as I mentioned, this function already existed here in manifest.c https://github.com/clearlinux/swupd-client/blob/master/src/manifest.c#L1049 and I made it a little bit more generic by receiving a list as an argument instead of a manifest, but I agree that now that it is more generic it is more appropriate to print the message from the place where the function is called. I'll make that change.

Reply:
done